src/backend/distributed/deparser/deparse_sequence_stmts.c

```diff
@@ -0,0 +1,158 @@
+ */
+
+#include "postgres.h"
+
+#include "catalog/namespace.h"
+#include "distributed/deparser.h"
+#include "utils/builtins.h"
+#include "utils/lsyscache.h"
+
+
+/* forward declaration for deparse functions */
+static void AppendDropSequenceStmt(StringInfo buf, DropStmt *stmt);
+static void AppendSequenceNameList(StringInfo buf, List *objects, ObjectType objtype);
+static void AppendRenameSequenceStmt(StringInfo buf, RenameStmt *stmt);
+
+/*
+ * DeparseDropSequenceStmt builds and returns a string representing the DropStmt
+ */
+char *
+DeparseDropSequenceStmt(Node *node)
+{
+	DropStmt *stmt = castNode(DropStmt, node);
+	StringInfoData str = { 0 };
+	initStringInfo(&str);
+
+	Assert(stmt->removeType == OBJECT_SEQUENCE);
+
+	AppendDropSequenceStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+/*
+ * AppendDropSequenceStmt appends a string representing the DropStmt to a buffer
+ */
+static void
+AppendDropSequenceStmt(StringInfo buf, DropStmt *stmt)
+{
+	appendStringInfoString(buf, "DROP SEQUENCE ");
+
+	if (stmt->missing_ok)
+	{
+		appendStringInfoString(buf, "IF EXISTS ");
+	}
+
+	AppendSequenceNameList(buf, stmt->objects, stmt->removeType);
+
+	if (stmt->behavior == DROP_CASCADE)
+	{
+		appendStringInfoString(buf, " CASCADE");
+	}
+
+	appendStringInfoString(buf, ";");
+}
+
+
+/*
+ * AppendSequenceNameList appends a string representing the list of sequence names to a buffer
+ */
+static void
+AppendSequenceNameList(StringInfo buf, List *objects, ObjectType objtype)
+{
+	ListCell *objectCell = NULL;
+	foreach(objectCell, objects)
+	{
+		if (objectCell != list_head(objects))
+		{
+			appendStringInfo(buf, ", ");
+		}
+
+		RangeVar *seq = makeRangeVarFromNameList((List *) lfirst(objectCell));
+
+		if (seq->schemaname == NULL)
+		{
+			Oid schemaOid = RangeVarGetCreationNamespace(seq);
+			seq->schemaname = get_namespace_name(schemaOid);
+		}
+
+		char *qualifiedSequenceName = quote_qualified_identifier(seq->schemaname,
+																 seq->relname);
+		appendStringInfoString(buf, qualifiedSequenceName);
+	}
+}
+
+
+/*
+ * DeparseRenameSequenceStmt builds and returns a string representing the RenameStmt
+ */
+char *
+DeparseRenameSequenceStmt(Node *node)
+{
+	RenameStmt *stmt = castNode(RenameStmt, node);
+	StringInfoData str = { 0 };
+	initStringInfo(&str);
+
+	Assert(stmt->renameType == OBJECT_SEQUENCE);
+
+	AppendRenameSequenceStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+/*
+ * AppendRenameSequenceStmt appends a string representing the RenameStmt to a buffer
+ */
+static void
+AppendRenameSequenceStmt(StringInfo buf, RenameStmt *stmt)
+{
+	RangeVar *seq = stmt->relation;
+
+	char *qualifiedSequenceName = quote_qualified_identifier(seq->schemaname,
+															 seq->relname);
+
+	appendStringInfoString(buf, "ALTER SEQUENCE ");
+
+	if (stmt->missing_ok)
+	{
+		appendStringInfoString(buf, "IF EXISTS ");
+	}
+
+	appendStringInfoString(buf, qualifiedSequenceName);
+
+	appendStringInfo(buf, " RENAME TO %s;", quote_identifier(stmt->newname));
```

Comment:
Super minor: Personally I kind of dislike when we add ; to a command, because it is only necessary when sending multiple queries as a single command (or to tell psql about the end of the command).

Reply:
I was following rename function style, so your preferred format is not followed consistently throughout the code. However, I also prefer your preferred format so will change it.